src/discussions/learners/LearnerPostsView.test.jsx

```diff
@@ -166,4 +169,57 @@ describe('Learner Posts View', () => {
       });
     });
   });
+
+  it.each([
+    { searchBy: 'type-all', result: 2 },
+    { searchBy: 'cohort-1', result: 2 },
+  ])('successfully display learners by %s.', async ({ searchBy, result }) => {
+    await setUpPrivilages(axiosMock, store, true);
+    axiosMock.onGet(getCohortsApiUrl(courseId))
+      .reply(200, Factory.buildList('cohort', 3));
+
+    await executeThunk(
+      fetchCourseCohorts(courseId),
+      store.dispatch,
+      store.getState,
+    );
+    await waitFor(() => {
+      renderComponent();
+    });
+
+    const filterBar = container.querySelector('.collapsible-trigger');
+    await act(async () => {
+      fireEvent.click(filterBar);
+    });
+
+    await waitFor(async () => {
+      const cohort = container.querySelector(`[for='${searchBy}']`);
+
+      await act(async () => {
+        fireEvent.click(cohort);
+      });
+      await waitFor(() => {
+        const learners = container.querySelectorAll('.discussion-post');
+
+        expect(learners).toHaveLength(result);
+      });
+    });
+  });
+
+  it('should display load more posts button and display more posts by clicking on button.', async () => {
+    await waitFor(() => {
+      renderComponent();
+    });
+    const loadMoreButton = container.querySelector('[data-testid="load-more"]');
```

Comment:
- wrap the waitFor(()=> (container.querySelector('[data-testid="load-more"]'))
- Remove waitFor from renderComponent();